ruskeyverbs/views.py

```diff
@@ -26,19 +51,618 @@ def index(request):
 
 @login_required
 def VerbListPerUser(request):
-    # get list of all available verbs sorted by due date for given user
-    verbs_with_due_dates = Verb.objects.annotate(due=Min('example__performanceperexample__due_date', filter=Q(example__performanceperexample__user_id=request.user.pk))).order_by('due').exclude(due=None)
-    verbs_without_due_dates = Verb.objects.annotate(due=Min('example__performanceperexample__due_date', filter=Q(example__performanceperexample__user_id=request.user.pk))).exclude(~Q(due=None))
-    sorted_verb_list = list(chain(verbs_with_due_dates, verbs_without_due_dates))
-    # sorted_verb_list = sorted(Verb.objects.all(),
-    #                           key=lambda a: a.get_earliest_due_date(request.user))
-    current_user = request.user
+    """
+    list view for verbs available in the system. Sorts by due date
+    (if studied previously) then frequency ranking
+    """
+    # two query sets because otherwise unstudied verbs would display before
+    # studied ones
+    verbs_with_due_dates = Verb.objects.annotate(
```

Comment:
Having to do this with two different queries still seems awkward to me. Would something like this guy is recommending be any better I wonder?

https://stackoverflow.com/questions/15121093/django-adding-nulls-last-to-query

Reply:
I actually found out Django has a built in option for this with an F() expression.

Reply:
Cool